src/log.ts

```diff
@@ -1,21 +1,149 @@
-import { handlers, LevelName, Logger, LogRecord } from "std/log/mod.ts";
+import {
+  HandlerOptions,
+  handlers,
+  LevelName,
+  Logger,
+  LogMode,
+  LogRecord,
+} from "std/log/mod.ts";
 import { format } from "std/datetime/mod.ts";
-import { isTesting } from "./_utils.ts";
+import { join } from "std/path/mod.ts";
+import dirs from "dirs/mod.ts";
+import { isTesting, mbToBytes } from "./_utils.ts";
+import { ErgomaticError } from "./error.ts";
+import { existsSync } from "std/fs/exists.ts";
+import { BufWriterSync } from "std/io/buf_writer.ts";
 
 function formatter({ loggerName, levelName, msg }: LogRecord) {
   const datetime = format(new Date(), "yyyy-MM-dd HH:mm:ss");
 
   return `[${datetime}][${loggerName}][${levelName}] ${msg}`;
 }
 
+function logsDir() {
+  const dataDir = dirs("data_local");
+
+  if (!dataDir) {
+    throw new ErgomaticError("Failed to find data directory");
+  }
+
+  return join(dataDir, "ergomatic");
+}
+
+interface RotatingFileHandlerOptions extends HandlerOptions {
+  filename: string;
+  mode?: LogMode;
+  maxBytes: number;
+  maxBackupCount: number;
+}
+
+/**
+ * `RotatingFileHandler` provided by deno has async `setup()` method
+ * Because of this we would need to refactor a lot of code to make it work as
+ * our loggers are currently created in constructors, etc.
+ *
+ * Create a sync version of `RotatingFileHandler` and use that instead.
+ * This is basically just `RotatingFileHandler` but using `Deno.*Sync` methods instead.
+ */
+class SyncRotatingFileHandler extends handlers.FileHandler {
```

Comment:
I hope we'll be able to remove this at some point, `RotatingFileHandler.setup` being async might be a deno issue: https://github.com/denoland/deno_std/issues/3538#issuecomment-1676942783